contracts/staking/query.go

```diff
@@ -0,0 +1,121 @@
+package staking
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-sdk/contracts/abis"
+	"github.com/0xPolygon/polygon-sdk/state/runtime"
+	"github.com/0xPolygon/polygon-sdk/types"
+	"github.com/umbracle/go-web3"
+	"github.com/umbracle/go-web3/abi"
+)
+
+var (
+	// staking contract address
+	AddrStakingContract = types.StringToAddress("1001")
+
+	// Gas limit used when querying the validator set
+	queryGasLimit uint64 = 100000
+)
+
+func extractResponse(method *abi.Method, returnValue []byte) (map[string]interface{}, error) {
+	decodedResults, err := method.Outputs.Decode(returnValue)
+	if err != nil {
+		return nil, err
+	}
+
+	results, ok := decodedResults.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("failed type assertion from decodedResults to map")
+	}
+
+	return results, nil
+}
+
+func DecodeValidators(method *abi.Method, returnValue []byte) ([]types.Address, error) {
+	extractedMap, extractErr := extractResponse(method, returnValue)
+	if extractErr != nil {
+		return nil, fmt.Errorf("unable to extract results, %v", extractErr)
```

Comment:
I'd say if you want to do any kind of string interpolation it's much more readable to use Errorf than Sprintf inside of an Errors.New

Reply:
I agree with @lazartravica, in the codebase we usually do 2 types of error manipulation:
* `errors.New` for errors that are regular string messages and don't need additional runtime data
* `fmt.Errorf` for errors that required additional runtime data relevant for the error

That's why I opted to use `fmt.Errorf`, since it has a cleaner interface than using `errors.New(fmt.Sprintf(...))`